tests/controllers/ControllerFactory.ts

```diff
@@ -17,90 +17,49 @@ import EventService from '../../services/EventService';
+    return new AdminController(storageService, userAccountService, attendanceService);
   }
 
   public static attendance(conn: Connection): AttendanceController {
-    if (!ControllerFactory.attendanceController) {
-      const attendanceService = new AttendanceService(conn.manager);
-      ControllerFactory.attendanceController = new AttendanceController(attendanceService);
-    }
-    return ControllerFactory.attendanceController;
+    const attendanceService = new AttendanceService(conn.manager);
+    return new AttendanceController(attendanceService);
   }
 
-  public static auth(conn: Connection): AuthController {
-    if (!ControllerFactory.authController) {
-      const userAccountService = new UserAccountService(conn.manager);
-      const userAuthService = new UserAuthService(conn.manager);
-      const emailService = new EmailService();
-      ControllerFactory.authController = new AuthController(userAccountService, userAuthService, emailService);
-    }
-    return ControllerFactory.authController;
+  public static auth(conn: Connection, emailService: EmailService): AuthController {
+    const userAccountService = new UserAccountService(conn.manager);
+    const userAuthService = new UserAuthService(conn.manager);
+    return new AuthController(userAccountService, userAuthService, emailService);
   }
 
   public static event(conn: Connection): EventController {
-    if (!ControllerFactory.eventController) {
-      const eventService = new EventService(conn.manager);
-      const storageService = new StorageService();
-      const attendanceService = new AttendanceService(conn.manager);
-      ControllerFactory.eventController = new EventController(eventService, storageService, attendanceService);
-    }
-    return ControllerFactory.eventController;
+    const eventService = new EventService(conn.manager);
+    const storageService = new StorageService();
+    const attendanceService = new AttendanceService(conn.manager);
+    return new EventController(eventService, storageService, attendanceService);
   }
 
   public static leaderboard(conn: Connection): LeaderboardController {
-    if (!ControllerFactory.leaderboardController) {
-      const userAccountService = new UserAccountService(conn.manager);
-      ControllerFactory.leaderboardController = new LeaderboardController(userAccountService);
-    }
-    return ControllerFactory.leaderboardController;
+    const userAccountService = new UserAccountService(conn.manager);
+    return new LeaderboardController(userAccountService);
   }
 
-  public static merchStore(conn: Connection): MerchStoreController {
-    if (!ControllerFactory.merchStoreController) {
-      const merchStoreService = new MerchStoreService(conn.manager);
-      ControllerFactory.merchStoreController = new MerchStoreController(merchStoreService);
-    }
-    return ControllerFactory.merchStoreController;
+  public static merchStore(conn: Connection, emailService = new EmailService()): MerchStoreController {
```

Comment:
Wouldn't always passing a mock instance clutter up the test files a bit, since not every test would require `EmailService` usage? One alternative is that we make the param optional, and if its undefined, we create a mock instance in the factory.

Reply:
I think it's fine, let's keep the interface simple.